frontend/app/src/pages/main/workers/managed-workers/$managed-worker/components/managed-worker-metrics.tsx

```diff
@@ -0,0 +1,244 @@
+import { queries } from '@/lib/api';
+import { useQuery } from '@tanstack/react-query';
+import {
+  ManagedWorker,
+  SampleStream,
+} from '@/lib/api/generated/cloud/data-contracts';
+import { Loading } from '@/components/ui/loading';
+import AreaChart, {
+  MetricValue,
+  format2Dec,
+  formatPercentTooltip,
+} from '@/components/molecules/brush-chart/area-chart';
+import { useMemo, useState } from 'react';
+import { useParentSize } from '@visx/responsive';
+import { Separator } from '@/components/ui/separator';
+import { GetCloudMetricsQuery } from '@/lib/api/queries';
+import { DateTimePicker } from '@/components/molecules/time-picker/date-time-picker';
+import { Button } from '@/components/ui/button';
+import { ArrowPathIcon } from '@heroicons/react/24/outline';
+
+export function ManagedWorkerMetrics({
+  managedWorker,
+}: {
+  managedWorker: ManagedWorker;
+}) {
+  const [beforeInput, setBeforeInput] = useState<Date | undefined>();
+  const [afterInput, setAfterInput] = useState<Date | undefined>();
+  const [queryParams, setQueryParams] = useState<GetCloudMetricsQuery>({
+    // default after is 1 day
+    after: new Date(Date.now() - 24 * 60 * 60 * 1000).toISOString(),
+  });
+  const [rotate, setRotate] = useState(false);
+
+  const getCpuMetricsQuery = useQuery({
+    ...queries.cloud.getManagedWorkerCpuMetrics(
+      managedWorker?.metadata.id || '',
+      queryParams,
+    ),
+    enabled: !!managedWorker,
+    refetchInterval: () => {
+      return 5000;
+    },
+  });
+
+  const getMemoryMetricsQuery = useQuery({
+    ...queries.cloud.getManagedWorkerMemoryMetrics(
+      managedWorker?.metadata.id || '',
+      queryParams,
+    ),
+    enabled: !!managedWorker,
+    refetchInterval: () => {
+      return 5000;
+    },
+  });
+
+  const getDiskMetricsQuery = useQuery({
+    ...queries.cloud.getManagedWorkerDiskMetrics(
```

Comment:
future: do we want to combine metrics into one query?

Reply:
I think eventually some metrics can take much longer to query, so might be nice to have these be separate